nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -222,6 +280,187 @@ impl NexusSaga for SagaSnapshotCreate {
 
 // snapshot create saga: action implementations
 
+async fn ssc_save_disk_gen(
+    sagactx: NexusActionContext,
+) -> Result<Generation, ActionError> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+
+    let (.., disk) = LookupPath::new(&opctx, &osagactx.datastore())
+        .project_id(params.project_id)
+        .disk_name(&params.create_params.disk.clone().into())
```

Comment:
It seems like we should resolve the disk name in the API request that creates the saga and then use the disk id here.  Otherwise, we could operate on a different disk than the user requested.  (That is, someone could issue the "create snapshot request for disk Foo", then have Foo get deleted and another Foo get created, and we'll wind up snapshotting the wrong Foo.)

Reply:
Good catch! fixed in 89d30cb5